src/moin/utils/edit_locking.py

```diff
@@ -0,0 +1,331 @@
+# Copyright: 2019 MoinMoin:RogerHaase
+# License: GNU GPL v2 (or any later version), see LICENSE.txt for details.
+
+"""
+Create sqlite3 data models for edit locking and saving drafts. Expected usage
```

Comment:
I have a django project which uses sqlite3 db in production and, sadly, it sucks.

I have to daily clean up my inbox from all the "Database is locked" exceptions which occur when there are too many parallel writers.

I tried to find a way to fix this, but it seems it is a design limitation and the only fix is to use a real db like postgresql.

Reply:
Certainly a concern and I worry about bugs. Using the Locust load test with 5 locusts trying to update Home, there are a lot of successful failed attempts to edit.

Do you use:

        self.con.isolation_level = 'EXCLUSIVE'
        self.con.execute('BEGIN EXCLUSIVE')

on your app? That is supposed to lock the sqlite database until a commit or close. I think at the OS level there are 10 tries 1 second apart, then a failure. It is on my list to add a check for such a failure.

Reply:
"there are a lot of successful failed attempts" - did you mean "a lot of failed attempts"?

in the django app, the ORM is talking to sqlite and I tried some tweaking, but it did not really get better.